libs/sdk-ui-theme-provider/src/ThemeProvider/ThemeProvider.tsx

```diff
@@ -27,6 +27,17 @@ export interface IThemeProviderProps {
      * component tree.
      */
     workspace?: string;
+
+    /**
+     * Theme object to use instead of one loaded from workspace
+     */
+    adhocTheme?: ITheme;
```

Comment:
I suggest to rename to just `theme`. Theme object to use. If not specified, provider will load theme from workspace.

Reply:
Seconded, in
https://github.com/gooddata/gooddata-ui-sdk/pull/779/commits/bce80e5a9c9db928a8e135ce4a259b2378cac9c8
https://github.com/gooddata/gooddata-ui-sdk/pull/779/commits/4eec1f73d7fe48aa8bf131bc360543e117233362
I will be doing the same for the backend and workspace contexts (hopefully, that will get merged soon).

Reply:
also, there is an interplay with the backend/workspace and contexts. please expand the docs to mention the expectations

Reply:
@lupko Which doc do you mean? We have this mention of Backend/WorkspaceProvider above in this interface comments:
export interface IThemeProviderProps {
    /**
     * Analytical backend, from which the ThemeProvider will obtain selected theme object
     *
     * If you do not specify instance of analytical backend using this prop, then you MUST have
     * BackendProvider up in the component tree.
     */
    backend?: IAnalyticalBackend;
    /**
     * Identifier of analytical workspace, from which the ThemeProvider will obtain the selected theme identifier
     *
     * If you do not specify workspace identifier, then you MUST have WorkspaceProvider up in the
     * component tree.
     */
    workspace?: string;